app/routes.js

```diff
@@ -102,7 +102,7 @@ module.exports.paths = paths
 
 module.exports.bind = function (app) {
   const account = new Router({ mergeParams: true })
-  account.use(getServiceAndAccount, userIsAuthorised)
+  account.use(getServiceAndAccount, userIsAuthorised, ensureSessionHasCsrfSecret, validateAndRefreshCsrf)
```

Comment:
Apologies I missed this before it was reviewed, I don't think you need `ensureSessionHasCsrfSecret` here, I believe that's done once when the session is initially authorised and created and should then be subsequently available -- I think the issue with the tests passing was with the payment types integration test